cli/src/cmd/forge/script/executor.rs

```diff
@@ -147,13 +146,23 @@ impl ScriptArgs {
     }
 
     /// Creates the Runner that drives script execution
-    async fn prepare_runner(&self, script_config: &ScriptConfig, sender: Address) -> ScriptRunner {
+    async fn prepare_runner(
+        &self,
+        script_config: &mut ScriptConfig,
+        sender: Address,
+    ) -> ScriptRunner {
         trace!("preparing script runner");
         let env = script_config.evm_opts.evm_env().await;
 
-        // the db backend that serves all the data
-        let db =
-            Backend::spawn(script_config.evm_opts.get_fork(&script_config.config, env.clone()));
+        // The db backend that serves all the data.
+        // We only call this function twice: when executing locally and when executing the
+        // broadcastable transactions.
+        let db = script_config.backend.take().unwrap_or_else(|| {
+            let backend =
+                Backend::spawn(script_config.evm_opts.get_fork(&script_config.config, env.clone()));
+            script_config.backend = Some(backend.clone());
+            backend
+        });
```

Comment:
I'm afraid this will only work for 2 times, right?

1. None -> create set `backend`
2. take `backend`
3. None ->  create set `backend`

right?

so instead of take it should probably be `cloned()`?

Reply:
yeah, but we only use it 2 times as i tried to allude in the comment. So this way we save a clone. If it changes in the future, we can change accordingly no?

Reply:
Ok actually it's not true... it can be called 3 times, if it goes through `rerun_with_new_deployer`

Reply:
cloning an empty Backend is no big deal, it's just a bunch of empty maps and empty DB

that's also what we do for every test we're running so redundant clone is fine